packages/core/src/Tooltip/index.tsx

```diff
@@ -248,37 +314,49 @@ export const Tooltip: React.FC<TooltipProps | ExpandedTooltipProps> = ({
           <PopOver
             anchorEl={anchorEl}
             horizontalPosition={horizontalLayout}
-            horizontalAlignment={horizontalLayout}
-            verticalPosition={hasOverflow ? 'top' : 'bottom'}
-            verticalAlignment={hasOverflow ? 'bottom' : 'top'}
+            horizontalAlignment={
+              placement === 'left-right'
+                ? horizontalAlign[horizontalLayout]
+                : horizontalLayout
+            }
```

Comment:
I still needed to check `placement` here because `horizontalLayout` can be `Left` when placement is "up-down" and there is no space on the left side.